Makefile

```diff
@@ -0,0 +1,63 @@
+# Define pkgs, run, and cover vairables for test so that we can override them in
+# the terminal more easily.
+pkgs := $(shell go list ./...)
+run := .
+count := 1
+
+## help: Show this help message
+help: Makefile
+	@echo " Choose a command run in "$(PROJECTNAME)":"
+	@sed -n 's/^##//p' $< | column -t -s ':' |  sed -e 's/^/ /'
+.PHONY: help
+
+## clean: clean testcache
+clean:
+	@echo "--> Clearing testcache"
+	@go clean --testcache
+.PHONY: clean
+
+## cover: generate to code coverage report.
+cover:
+	@echo "--> Generating Code Coverage"
+	@go install github.com/ory/go-acc@latest
+	@go-acc -o coverage.txt $(pkgs)
+.PHONY: cover
+
+## deps: Install dependencies
+deps:
+	@echo "--> Installing dependencies"
+	@go mod download
+#	@make proto-gen
+	@go mod tidy
+.PHONY: deps
+
+## lint: Run linters golangci-lint and markdownlint.
+lint: vet
```

Comment:
we should add `actionlint` too for linting the github action files. 
i'll make sure that is in the template repo. 

https://github.com/rhysd/actionlint

Reply:
Should this be a part of Makefile, or directly in GH action?

Reply:
it has to be in the Makefile because it is linting the action file itself. If there are lint errors the action won't run so it is actually impossible to run in the action file itself, on itself. it would only lint the other files. 
So keeping it locally makes the most sense.

Reply:
Created follow-up issue: https://github.com/rollkit/based-sequencer/issues/11